astropy/table/__init__.py

```diff
@@ -59,3 +59,8 @@ class Conf(_config.ConfigNamespace):
     from astropy.io.fits import connect
     from astropy.io.misc import connect
     from astropy.io.votable import connect
+    try:
```

Comment:
Related to the other comment about when to import, the precedent is to always register all of the core readers even if optional dependencies are not available.  I can see arguments for going either way on this, but in some sense it is good to advertise the availability of a format (ASDF in particular) in the unified interface.  Just like you can get docstrings for functions or methods even if they have dependencies that are not installed.

So the idea would be to remove the try/except at this point.

Reply:
In theory you could put the connect import at: https://github.com/astropy/astropy/blob/master/astropy/io/misc/connect.py#L5

I think it is more obvious if it lives in the table init rather than hidden in `io.misc.connect`, but I don't feel strongly either way.

Reply:
This makes sense.